### PR TITLE
Bugfix

### PR DESCRIPTION
**Pull request by kunyuan**
1. add support for the keyword "verbose", which is more standard than "print". 
2. function to create multi-dimensional Continuous/Discrete variables.
3. Automatically adjust the size of variable pools.
4. Now Results always return vector
5. update readme

### CODE REVIEW

File: src/configuration.jl [hunk 1]
```diff
@@ -1,31 +1,38 @@
...
+- `rng`: a MersenneTwister random number generator, seeded by `seed`
+- `userdata`: user-defined parameter
+- `var`: TUPLE of variables, each variable should be derived from the abstract type Variable, see variable.jl for details). Use a tuple rather than a vector improves the performance.
 
- ## integrand properties
+# integrand properties
 - `neighbor::Vector{Tuple{Int, Int}}` : vector of tuples that defines the neighboring integrands. Two neighboring integrands are directly connected in the Markov chain. 
     e.g., [(1, 2), (2, 3)] means the integrand 1 and 2 are neighbor, and 2 and 3 are neighbor.  
    The neighbor vector defines a undirected graph showing how the integrands are connected. Please make sure all integrands are connected.
    By default, we assume the N integrands are in the increase order, meaning the neighbor will be set to [(N+1, 1), (1, 2), (2, 4), ..., (N-1, N)], where the first N entries are for diagram 1, 2, ..., N and the last entry is for the normalization diagram. Only the first diagram is connected to the normalization diagram.
```
Comment (houpc) [inline src/configuration.jl:23]: (2, 4) -> (2, 3)

File: README.md [hunk 2]
```diff
@@ -1,272 +1,144 @@
...
 ```
-This approach simplifies the evaluation of high-dimensional integrals involving multiple symmetric variables. For example, to calculate the area of a quarter unit circle (π/4 = 0.785398...):
+## Quick Start
+MCIntegration.jl simplifies complex integral calculations. Here are two examples to get you started.
+
+To estimate the integral $\int_0^1 \frac{\log(x)}{\sqrt{x}} dx = 4$, you can use:
 ```julia
-julia> res = integrate((x, c)->(x[1]^2+x[2]^2<1.0); var = x, dof = 2) 
-Integral 1 = 0.7860119307731648 ± 0.002323473435947719   (chi2/dof = 2.14)
+julia> f(x, _) = log(x[1]) / sqrt(x[1])   # Define your integrand where 'x' represents the random variable in the integral
```
Comment (houpc) [inline README.md:23]: Use f(x, c) to be consistent with the following "an integrand function `f(x, c)`"

File: README.md [hunk 3]
```diff
@@ -1,272 +1,144 @@
...
 
-# Variables
+Given its robustness and efficiency, the default solver in this package is the `:vegasmc`. To choose a specific solver, use the `solver` parameter in the `integrate` function, like `solver=:vegas`.
 
-The package supports a couple of common types random variables. You can create them using the following constructors,
+Please note that the calling convention for the user-defined integrand for `:mcmc` is slightly different from that of `:vegas` and `:vegasmc`. Please refer to the separate detailed note on this.
 
-- `Continous(lower, upper[; adapt = true, alpha = 3.0, ...])`: Continuous real-valued variables on the domain [lower, upper). MC will learn the distribution using the Vegas algorithm and then perform an imporant sampling accordingly.
-- `Discrete(lower::Int, upper::Int[; adapt = true, alpha = 3.0, ...])`: Integer variables in the closed set [lower, upper]. MC will learn the distribution and perform an imporant sampling accordingly.
+Packed variables can enhance the efficiency of :vegasmc and :mcmc solvers by reducing the auto-correlation time of the Markov chain, leading to a more effective sampling proces
```
Comment (houpc) [inline README.md:91]: ... sampling process.

File: docs/src/index.md [hunk 4]
```diff
@@ -38,40 +49,55 @@ ignore        -3.8394711 ± 0.12101621              -3.8394711 ± 0.12101621
     10        -3.9999099 ± 0.0033455927            -3.9979808 ± 0.0013607691               1.9269
 -------------------------------------------------------------------------------------------------
 ```
-- By default, the function performs 10 iterations and each iteraction costs about `1e4` evaluations. You may reset these values with `niter` and `neval` keywords arguments.
+- By default, the function performs 10 iterations and each iteraction costs about `1e4` evaluations. You can adjust these values using `niter` and `neval` keywords arguments.
```
Comment (houpc) [inline docs/src/index.md:52]: typo：iteraction  -> iteration

File: docs/src/index.md [hunk 5]
```diff
@@ -92,7 +118,38 @@ julia> integrate(var = Continuous(0.0, 1.0), dof = [[2,], [3,]], inplace=true) d
        end
 ```
 
-## Measure Histogram
+### Example 5. Use `Configuration` to Interface with MCIntegration 
+
+- `Configuration` in integrands: As explained in the Example 1, the user-defined integrand has the signature `(x, c)` where `x` is the variable(s), and `c` is a ['Configuration'](https://numericaleft.github.io/MCIntegration.jl/dev/lib/montecarlo/#Main-module) struct stores the essential state information for the Monte Carlo sampling.Three particularly relavent members of `Configuratoin` include
```
Comment (houpc) [inline docs/src/index.md:123]: Typo: `Configuratoin` -> `Configuration`

**Review by houpc — APPROVED** [review]

File: README.md [hunk 6]
```diff
@@ -1,272 +1,144 @@
...
 
-The packed variables will be sampled all together in the Markov-chain based solvers (`:vegasmc` and `:mcmc`). Such updates will generate more independent samples compared to the unpacked version. Sometimes, it could reduce the auto-correlation time of the Markov chain and make the algorithm more efficient.
+  **Note:** For MPI functionality, install [MPI.jl](https://github.com/JuliaParallel/MPI.jl) package and follow the [configuration](https://juliaparallel.github.io/MPI.jl/stable/configuration/) instructions.
 
-Moreover, packed variables usually indicate nontrivial correlations between their distributions. In the future, it will be interesting to learn such correlation so that one can sample the packed variables more efficiently.
+   
 
-# Parallelization
-MCIntegration supports both MPI and multi-thread parallelization. You can even mix them if necessary.
+- Multi-threading
```
Comment (iintSjds) [inline README.md:109]: **Multi-threading**

**Pull request merged**